src/envoy/http/authn/http_filter.cc

```diff
@@ -14,11 +14,38 @@
  */
 
 #include "src/envoy/http/authn/http_filter.h"
+#include "common/http/utility.h"
 #include "src/envoy/http/authn/mtls_authentication.h"
 #include "src/envoy/utils/utils.h"
 
 namespace Envoy {
 namespace Http {
+namespace {
+
+bool MatchCredentialRule(const std::string& peer_user,
+                         const iaapi::CredentialRule& rule) {
+  if (rule.matching_peers_size() == 0) {
+    return true;
+  }
+  for (const auto& allowed_id : rule.matching_peers()) {
+    if (peer_user == allowed_id) {
```

Comment:
We need to optimize this.  Remember we are in data-path.  Speed is very important in data-path

Reply:
Noted. I like to defer this to see how all the jwt-auth caching work together.

On the other hand, note the typical usage doesn't have large set of matching peers (my guess is 1 or 2 on average). So any pre-processing/ changing data structure does not guarantee a better performance.